UI/Web/src/app/book-reader/_components/book-reader/book-reader.component.ts

```diff
@@ -851,41 +856,78 @@ export class BookReaderComponent implements OnInit, AfterViewInit, OnDestroy {
+    const height = this.windowHeight - (this.topOffset * 2)
+    let maxHeight = 'unset'
+    let maxWidth = ''
+    switch (this.layoutMode) {
+      case BookPageLayoutMode.Default:
+        if (isVerticalWritingStyle) {
+          maxHeight = `${height}px`
+        } else {
+          maxWidth = `${this.getVerticalPageWidth()}px`
+        }
+        break
+
+      case BookPageLayoutMode.Column1:
+        maxHeight = `${height}px`
+        maxWidth = `${this.getVerticalPageWidth()}px`
+        break
+
+      case BookPageLayoutMode.Column2:
+        maxWidth = `${this.getVerticalPageWidth()}px`
+        if (isVerticalWritingStyle && !this.isSingleImagePage)  {
+          maxHeight = `${height / 2}px`
+        } else {
+          maxHeight = `${height}px`
+        }
+        break
+    }
+    this.document.documentElement.style.setProperty('--book-reader-content-max-height', maxHeight)
+    this.document.documentElement.style.setProperty('--book-reader-content-max-width', maxWidth)
+
+  }
+
+  updateSingleImagePageStyles() {
+    if (this.isSingleImagePage && this.layoutMode !== BookPageLayoutMode.Default) {
+      this.document.documentElement.style.setProperty('--book-reader-content-position', 'absolute');
+      this.document.documentElement.style.setProperty('--book-reader-content-top', '50%');
+      this.document.documentElement.style.setProperty('--book-reader-content-left', '50%');
+      this.document.documentElement.style.setProperty('--book-reader-content-transform', 'translate(-50%, -50%)');
     } else {
-      maxHeight = undefined;
+        this.document.documentElement.style.setProperty('--book-reader-content-position', '');
+        this.document.documentElement.style.setProperty('--book-reader-content-top', '');
+        this.document.documentElement.style.setProperty('--book-reader-content-left', '');
+        this.document.documentElement.style.setProperty('--book-reader-content-transform', '');
     }
-    Array.from(images).forEach(img => {
-      if (maxHeight === undefined) {
-        this.renderer.removeStyle(img, 'max-height');
-      } else if (this.writingStyle === WritingStyle.Horizontal) {
-        this.renderer.setStyle(img, 'max-height', `${maxHeight}px`);
-      } else {
-        const aspectRatio = img.width / img.height;
-        const pageWidth = this.getVerticalPageWidth()
-        const maxImgHeight = Math.min(maxHeight, pageWidth / aspectRatio);
-        this.renderer.setStyle(img, 'max-height', `${maxImgHeight}px`);
-      }
-    });
+  }
 
+  checkSingleImagePage(safeHtml: SafeHtml) {
+    const parser = new DOMParser();
+    const doc = parser.parseFromString(safeHtml.toString(), "text/html");
+    const images = doc.querySelectorAll("img");
+    const text = doc.querySelectorAll("p");
+
+
+    if (images.length > 1 || text.length > 1) {
+      return false
+    }
+    // A p tag could hold the image, and needs to be checked
+    if (text.length) {
```

Comment:
What if the image is within a div, we might want to expand the tags.

Reply:
You raise a valid point. Forgot that there could be books that have their text inside of "div"s instead of "p"-tags or other kinds of elements. I will re-do this section to handle these scenarios

Reply:
Once this is sorted, I'm ready to merge.